src/v2/System/Router/ErrorBoundary.tsx

```diff
@@ -43,59 +51,59 @@ export class ErrorBoundary extends React.Component<Props, State> {
      */
     if (error.message.match(/Loading chunk .* failed/)) {
       return {
+        isError: true,
         asyncChunkLoadError: true,
-        errorStack,
+        detail,
+        message,
       }
     }
 
     return {
+      isError: true,
       genericError: true,
-      errorStack,
+      detail,
+      message,
     }
   }
 
   render() {
-    const { asyncChunkLoadError, genericError, errorStack } = this.state
+    const {
+      asyncChunkLoadError,
+      detail,
+      genericError,
+      isError,
+      message,
+    } = this.state
 
-    switch (true) {
-      case asyncChunkLoadError: {
-        return (
-          <ErrorModalWithReload
-            message="Please check your network connection and try again."
-            show={asyncChunkLoadError}
-          />
```

Comment:
This error modal is really nice in different contexts, particularly on mobile. As long as the reload functionality is there and very clear to the end user we can go with something more generic (`<ErrorPage>`)

Reply:
To elaborate further with a better example, the modal with reload is helpful for when there are page timeouts (often with mobile), and this modal appears, saying things have timed out, and then there's a reload button that the user can click

Reply:
Ah, i see we still have the network timeout [here](https://github.com/artsy/force/pull/9153/files#diff-45adeffd54fe16decbf8e730d662c0a6569d769b0566d2016b3b22f894bd76ccR8). This error is a bit less worrisome, though a 'reload page' button would still be nice UX if we see exactly the `asyncChunkLoadError` error since a page reload tends to fix it.